dashboard/app/models/teacher_score.rb

```diff
@@ -56,12 +56,13 @@ def self.score_level_for_student(
     )
   end
 
-  def self.get_level_scores_for_script_for_section(script_id, section_id)
+  def self.get_level_scores_for_script_for_section(script_id, section_id, page)
     level_scores_by_student_by_stage_by_script = {}
     stages = Script.find(script_id).stages
+    student_ids = Section.find(section_id).students.pluck(:id).each_slice(50).to_a[page - 1]
```

Comment:
A tiny optimization: As written, I think this query retrieves all the student ids from the section for every request, and then slices them.  That's probably not too expensive, but we have a gem called [`kaminari`](https://github.com/kaminari/kaminari) that adds pagination features to the ActiveRecord query interface.  Here, I _think_ you'd do something like this, which should use MySQL `LIMIT` and `OFFSET` features to only ask the database for 50 ids (described in MySQL's [`SELECT` documentation](https://dev.mysql.com/doc/refman/5.7/en/select.html)).

```rb
student_ids = Section.find(section_id).students.page(page).per(50).pluck(:id)
```

There might be a very small benefit in readability too, if we standardize on this approach for paginated queries.